cmd/check.go

```diff
@@ -0,0 +1,318 @@
+	postgres, conn, dsn, err := setupDatabase(ctx, tmpDir, "check", 5434)
+	defer func() {
+		if conn != nil {
+			_ = conn.Close(ctx)
+		}
+		if postgres != nil {
+			_ = postgres.Stop()
+		}
+	}()
+	if err != nil {
+		return fmt.Errorf("failed to setup database: %w", err)
+	}
+	dsn = fmt.Sprintf("%s?sslmode=disable", dsn)
+
+	for _, u := range config.DatabaseUsers {
+		var userExists bool
+		userExists, err = internal.CheckUserExists(ctx, conn, u)
+		if err != nil {
+			return fmt.Errorf("failed to check if user exists: %w", err)
+		}
+		if !userExists {
+			_, err = conn.Exec(ctx, fmt.Sprintf("CREATE ROLE %q WITH LOGIN PASSWORD 'postgres'", u))
+			if err != nil {
+				return fmt.Errorf("failed to create user: %w", err)
+			}
+		}
+	}
+
+	migrationFiles, err := internal.FindMigrations(migrationsDir, true)
+	if err != nil {
+		return fmt.Errorf("failed to find migrations: %w", err)
+	}
+
+	hookOptions := &internal.HookOptions{
+		Env: map[string]string{
+			"TREK_POSTGRES_HOST":     "localhost",
+			"TREK_POSTGRES_PORT":     "5434",
+			"TREK_POSTGRES_USER":     "postgres",
+			"TREK_POSTGRES_PASSWORD": "postgres",
+			"TREK_POSTGRES_DATABASE": "postgres",
+			"TREK_POSTGRES_SSLMODE":  "disable",
+		},
+	}
+
+	err = internal.RunHook(wd, "check-pre", hookOptions)
+	if err != nil {
+		return fmt.Errorf("failed to run hook: %w", err)
+	}
+
+	log.Println("Checking dbm file")
+
+	err = checkDBM(config, wd)
+	if err != nil {
+		return fmt.Errorf("failed to check dbm: %w", err)
+	}
+
+	log.Println("Checking migration file names")
+
+	err = checkMigrationFileNames(migrationFiles)
+	if err != nil {
+		return fmt.Errorf("failed to check migration file names: %w", err)
+	}
+
+	log.Println("Checking templates")
+
+	err = checkTemplates(config, uint(len(migrationFiles)))
+	if err != nil {
+		return fmt.Errorf("failed to check templates: %w", err)
+	}
+
+	log.Println("Checking migrations and testdata")
+
+	err = checkMigrationsAndTestdata(wd, migrationsDir, dsn, migrationFiles)
+	if err != nil {
+		return fmt.Errorf("failed to check migrations and testdata: %w", err)
+	}
+
+	for _, u := range config.DatabaseUsers {
+		_, err = conn.Exec(ctx, fmt.Sprintf("GRANT SELECT ON public.schema_migrations TO %q", u))
+		if err != nil {
+			return fmt.Errorf("failed to grant select permission on schema_migrations to %q: %w", u, err)
+		}
+	}
+
+	err = internal.RunHook(wd, "check-post", hookOptions)
+	if err != nil {
+		return fmt.Errorf("failed to run hook: %w", err)
+	}
+
+	return nil
+}
+
+func checkDBM(config *internal.Config, wd string) error {
+	model := dbm.DBModel{}
+
+	m, err := os.ReadFile(filepath.Join(wd, fmt.Sprintf("%s.dbm", config.ModelName)))
+	if err != nil {
+		return fmt.Errorf("failed to read model file: %w", err)
+	}
+
+	err = xml.Unmarshal(m, &model)
+	if err != nil {
+		return fmt.Errorf("failed to parse model: %w", err)
+	}
+
+	modelRoles := map[string]struct{}{}
+	for _, role := range model.Roles {
+		if !role.SQLDisabled {
+			//nolint:goerr113
```

Comment:
Why is this required? (the nolint rules)

Reply:
this specific linter has this behaviour which we don't want, but other behaviour that we want (like error comparison using `errors.Is(...)`. that's why it's ignored.